.changeset/thirty-swans-fail.md

```diff
@@ -0,0 +1,5 @@
+---
+"react-router": patch
+---
+
+Resolve relative paths correctly in hash routes that are children of pathless routes (#11629)
```

Comment:
```suggestion
When using `v7_relativeSplatPath`, properly resolve relative paths in splat routes that are children of pathless routes
```

The pull request number will be added automatically during the release process, and the PR will link to the issue number :)